### PR TITLE
SNOW-802269-Jsontuple-cbrt-fromjson-datesub-trycast

### PR DESCRIPTION
**Pull request by sfc-gh-sjayabalan**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

Added following functions in java and scala.
a. json_tuple()
b.from_json()
c.date_sub()
d.try_cast()
e.cbrt()

2. Fill out the following pre-review checklist:

   - [ yes] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

  Added following functions as wrapper to java and scala functions.

## Pre-review checklist

(For Snowflake employees)

- [ ] This change has passed precommit
- [ ] I have reviewed code coverage report for my PR in  ([Sonarqube](https://sonarqube.int.snowflakecomputing.com/dashboard?id=snowpark))

### CODE REVIEW

File: src/main/java/com/snowflake/snowpark_java/Functions.java [hunk 1]
```diff
@@ -4641,6 +4641,222 @@ public static Column format_number(Column x, Integer d) {
...
+    int i = -1;
+    java.util.ArrayList<com.snowflake.snowpark_java.Column> result =
+        new java.util.ArrayList<Column>();
+    for (int j = 0; j < fields.length; j++) {
+      i = i + 1;
+      result.add(Functions.callUDF("JSON_EXTRACT_PATH_TEXT", json, col(fields[j])).as("c" + i));
+    }
+    return result;
+  }
+
```
Comment (sfc-gh-bli) [inline src/main/java/com/snowflake/snowpark_java/Functions.java:4701]: missing doc.

File: src/main/scala/com/snowflake/snowpark/functions.scala [hunk 2]
```diff
@@ -3887,6 +3887,256 @@ object functions {
...
+   * @return The result is of type targetType.
+   *  special version of CAST for a subset of datatype conversions.
+   * It performs the same operation
+   * (i.e. converts a value of one data type into another data type),
+   * but returns a NULL value instead of raising an error
+   * when the conversion can not be performed.
+   * The column argument must be a string column in Snowflake.
+   */
+  def try_cast(e: Column, targetType: DataType): Column = {
+    e.cast(targetType)
```
Comment (sfc-gh-mrojas) [inline src/main/scala/com/snowflake/snowpark/functions.scala:4108]: Do you have tests cases to validate that this returns null if cast fails? I think this might fail and that you might need to use try_cast built int

File: src/test/java/com/snowflake/snowpark_test/JavaFunctionSuite.java [hunk 3]
```diff
@@ -3186,4 +3188,81 @@ public void to_utc_timestamp() {
     Row[] expected = {Row.create(Timestamp.valueOf("2024-04-05 01:02:03.0"))};
     checkAnswer(df.select(Functions.to_utc_timestamp(df.col("a"))), expected, false);
   }
+
+  @Test
+  public void json_tuple1() {
```
Comment (sfc-gh-mrojas) [inline src/test/java/com/snowflake/snowpark_test/JavaFunctionSuite.java:3193]: use a more representative test name like json_tuple_<something that indicates what is done>

maybe json_tuple_single_parameter ? for example

File: src/test/java/com/snowflake/snowpark_test/JavaFunctionSuite.java [hunk 4]
```diff
@@ -3186,4 +3188,81 @@ public void to_utc_timestamp() {
...
+      Row.create((Object) null),
+      Row.create("foo"),
+      Row.create((Object) null),
+      Row.create((Object) null)
+    };
+    checkAnswer(jsontupleDF, expected, false);
+  }
+
+  @Test
+  public void json_tuple2() {
```
Comment (sfc-gh-mrojas) [inline src/test/java/com/snowflake/snowpark_test/JavaFunctionSuite.java:3213]: Same as with preivous

File: src/test/java/com/snowflake/snowpark_test/JavaFunctionSuite.java [hunk 5]
```diff
@@ -3186,4 +3188,81 @@ public void to_utc_timestamp() {
...
+        df.select(
+            (Functions.json_tuple(Functions.col("v"), Functions.col("id"), Functions.col("name"))
+                .toArray(new Column[0])));
+    jsontupleDF.show();
+    Row[] expected = {Row.create(("5"), ("Jose"))};
+    checkAnswer(jsontupleDF, expected, false);
+  }
+
+  @Test
+  public void try_cast() {
```
Comment (sfc-gh-mrojas) [inline src/test/java/com/snowflake/snowpark_test/JavaFunctionSuite.java:3230]: need more test cases. You need to test several combinarions

df = spark.createDataFrame(
     [(2, "123"), (5, "Bob"), (3, None)], ["age", "name"])
df.select(df.name.try_cast(LongType())).show()
+----+
|name|
+----+
| 123|
|NULL|
|NULL|
+----+
Structure types should be consider as well:

df.filter(F.array_contains(F.lit('Key'), F.col('KEY_UNSTRCUTURED_ARRAY ').cast('ARRAY(VARCHAR)')).show()
# or
df.filter(F.array_contains(F.lit('Key'), F.col('KEY_UNSTRCUTURED_ARRAY ').cast(T.ArraryType(T.StringType()))).show()

File: src/test/java/com/snowflake/snowpark_test/JavaFunctionSuite.java [hunk 6]
```diff
@@ -3186,4 +3188,81 @@ public void to_utc_timestamp() {
...
+    DataFrame df = getSession().sql("select * from values('2024-04-05') as t(a)");
+    Row[] expected = {Row.create(Date.valueOf("2024-04-05"))};
+    checkAnswer(df.select(Functions.try_cast(df.col("a"), DateType)), expected, false);
+  }
+
+  @Test
+  public void date_sub() {
+    DataFrame df =
```
Comment (sfc-gh-mrojas) [inline src/test/java/com/snowflake/snowpark_test/JavaFunctionSuite.java:3238]: We need more tests, what happens with null, what happens with large values

File: src/test/java/com/snowflake/snowpark_test/JavaFunctionSuite.java [hunk 7]
```diff
@@ -3186,4 +3188,81 @@ public void to_utc_timestamp() {
...
+                    + "('2020-08-21 01:30:05.000' :: timestamp) as T(a)");
+    Row[] expected = {
+      Row.create(Date.valueOf("2020-04-30")), Row.create(Date.valueOf("2020-08-20"))
+    };
+    checkAnswer(df.select(Functions.date_sub(df.col("a"), 1)), expected, false);
+  }
+
+  @Test
+  public void from_json() {
+    DataFrame df =
```
Comment (sfc-gh-mrojas) [inline src/test/java/com/snowflake/snowpark_test/JavaFunctionSuite.java:3251]: positive an negative tests needed.
